fenix/app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -1175,6 +1183,28 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
         navHost.navController.navigate(action)
     }
 
+    private fun showExtensionProcessDisabledDialog() {
+        val message = getString(R.string.addon_process_crash_dialog_message, appName)
+
+        val alertDialogBuilder = AlertDialog.Builder(this@HomeActivity, R.style.DialogStyleNormal).apply {
+            setCancelable(false)
+            setTitle(R.string.addon_process_crash_dialog_title)
+            setMessage(message)
+            setPositiveButton(R.string.addon_process_crash_dialog_retry_button_text
+            ) { dialog, _ ->
+                components.core.geckoRuntime.webExtensionController.enableExtensionProcessSpawning()
```

Comment:
Nit: **Please don't address this now, let's file a follow-up ticket, this is not required now it's just point of future improvement**.

We normally do not access specific `gecko-runtime` functionality directly, as it breaks the encapsulation of web engines, what we normally do is expose that behaviour via the [Engine](https://github.com/mozilla-mobile/firefox-android/blob/main/android-components/components/concept/engine/src/main/java/mozilla/components/concept/engine/Engine.kt), `GeckoEngine` or any `engine` so we are engine agnostic. As an example you could  see all the web extension functionalities on the [GeckoEngine](https://github.com/mozilla-mobile/firefox-android/blob/facb1a4b73ed48f54b46728bd0363119017af249/android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngine.kt#L223-L251).